app/lib/redis_configuration.rb

```diff
@@ -39,7 +39,38 @@ def namespace
   end
 
   def url
-    ENV['REDIS_URL']
+    if ENV['REDIS_SENTINEL']
+      m = ENV.fetch('REDIS_SENTINEL_MASTER', 'mymaster')
+      p = ENV.fetch('REDIS_PASSWORD', '')
+      "redis://:#{p}@#{m}"
+    else
+      ENV['REDIS_URL']
+    end
+  end
+
+  def sentinels
+    return unless ENV['REDIS_SENTINEL']
+
+    unless ENV['REDIS_SENTINEL'].include? ','
+      ips = Resolv.getaddresses(ENV['REDIS_SENTINEL'])
+      port = ENV.fetch('REDIS_SENTINEL_PORT', '26379')
+
+      ENV['REDIS_SENTINEL'] = ips.map do |ip|
+        "#{ip}:#{port}"
+      end.join(',')
+    end
+    ENV['REDIS_SENTINEL'].split(',').map do |server|
+      host, port = server.split(':')
+      { host: host, port: port.to_i }
+    end
```

Comment:
If I understand it correctly, you want to support either

1. specifying a single sentinel using `ENV['REDIS_SENTINEL']` in which case you can give the port number as `ENV['REDIS_SENTINEL_PORT']` **or**
2. specify a comma-seperated list of sentinels in `ENV['REDIS_SENTINEL']` in which case the port should be included in each entry (`host:port`)

Did I get that right?

If yes, why support two so very different ways of configuring sentinels? Could we not just use the latter?

I believe that is also what the JS version expects, is it not?

Also, only the first variant falls back to the default port. I believe the second (and possibly only one) should as well. Again, that is what the JS version does afaict.

Reply:
(Small addition: I see the same logic in `lib/mastodon/redis_config.rb`. The same reasoning applies I think, but that also means that at this point _here_ we should never ever encounter case 1. So I do not see a need to duplicate this logic here.)